mountpoint-s3/src/build_info.rs

```diff
@@ -9,18 +9,38 @@ mod built {
 
 /// Valid SemVer version constructed using declared Cargo version and short commit hash if needed.
 pub const FULL_VERSION: &str = {
+    const BUILD_TARGET_STR: &str = match build_target_suffix() {
+        Some(target) => target,
+        None => "",
+    };
+
     if is_official_aws_release() {
-        built::PKG_VERSION
+        if build_target_suffix().is_some() {
+            const_format::concatcp!(built::PKG_VERSION, "-", BUILD_TARGET_STR)
+        } else {
+            built::PKG_VERSION
+        }
+    } else if build_target_suffix().is_some() {
+        const_format::concatcp!(
+            built::PKG_VERSION,
+            "-",
+            BUILD_TARGET_STR,
+            "-unofficial",
+            git_commit_suffix()
+        )
     } else {
         const_format::concatcp!(built::PKG_VERSION, "-unofficial", git_commit_suffix())
     }
 };
-
-/// Checks environment to see if this build is for an official Mountpoint for Amazon S3 release.
 const fn is_official_aws_release() -> bool {
     option_env!("MOUNTPOINT_S3_AWS_RELEASE").is_some()
 }
 
+/// Gets build target from BUILD_TARGET environment variable if set
+const fn build_target_suffix() -> Option<&'static str> {
+    option_env!("BUILD_TARGET")
```

Comment:
np: How about `MOUNTPOINT_S3_DISTRIBUTION_METHOD` or `MOUNTPOINT_S3_PACKAGE_PLATFORM` to better distinguish between the generic existing builds (DEB/RPM/TAR.gz) and platform-specific packaging builds (AL2023 native packaging/distribution).